opengever/maintenance/scripts/fix_dossier_resolvability.py

```diff
@@ -0,0 +1,77 @@
+"""
+Provides functionality to fix the resolvability of a dossier with already
+resolved subdossiers that are now in an 'invalid' state.
+Invalid means that some of the documents in a dossier have a document date
+outside its dossiers start-end range. This most likely happens when we generate
+a journal pdf while closing a dossier with an already defined end-date.
+
+    bin/instance run ./scripts/fix_dossier_resolvability.py <path>
+
+"""
+from opengever.document.behaviors.metadata import IDocumentMetadata
+from opengever.dossier.behaviors.dossier import IDossier
+from opengever.maintenance.debughelpers import setup_app
+from opengever.maintenance.debughelpers import setup_option_parser
+from opengever.maintenance.debughelpers import setup_plone
+from plone import api
+import transaction
+
+
+def fix_dossier_resolvability(plone, path):
+    dossier = plone.unrestrictedTraverse(path)
+    if not dossier:
+        print "Could not find dossier at {}".format(path)
+        sys.exit(1)
+
+    catalog = api.portal.get_tool('portal_catalog')
+    query = {
+        'path': '/'.join(dossier.getPhysicalPath()),
+        'portal_type': 'opengever.document.document'
+    }
+
+    for brain in catalog(**query):
```

Comment:
This should probably be `catalog.unrestrictedSearchResults(**query)` - we might potentially have to deal with trashed documents too.

Reply:
AFAIK the validation rule queries restricted also. But you're right i can't hurt, i'll amend.